cmake/tools.cmake

```diff
@@ -90,22 +91,36 @@ endif ()
 
 if (LINKER_NAME)
     if (COMPILER_CLANG)
-        find_program (LLD_PATH NAMES ${LINKER_NAME})
-        if (NOT LLD_PATH)
-            message (FATAL_ERROR "Using linker ${LINKER_NAME} but can't find its path.")
-        endif ()
-
-        # This a temporary quirk to emit .debug_aranges with ThinLTO
-        set (LLD_WRAPPER "${CMAKE_CURRENT_BINARY_DIR}/ld.lld")
-        configure_file ("${CMAKE_CURRENT_SOURCE_DIR}/cmake/ld.lld.in" "${LLD_WRAPPER}" @ONLY)
+        if (LINKER_NAME MATCHES "lld")
+            find_program (LLD_PATH NAMES ${LINKER_NAME})
```

Comment:
`find_program()` + the check for the path (l. 94 - 96 on the left side) were IMHO a bit misleading. This code was supposed to be a generic check for *all* linkers, not only lld. Therefore, suggest to do this:

```cmake
if (COMPILER_CLANG)
    find_program (LINKER_FOUND ${LINKER_NAME})
    if (NOT LINKER_FOUND)
        message (FATAL_ERROR "Using linker ${LINKER_NAME} but can't find its path.")
    endif ()

    if (LINKER_NAME MATCHES "lld")
        # << here, configure temporary quirk to emit .debug_aranges with ThinLTO >>
    endif()

    set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} --ld-path=${LINKER_FOUND}")
    set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} --ld-path=${LINKER_FOUND}")
```